--- policy-1.22/file_contexts/program/mta.fc.orig	2005-02-16 20:40:16.000000000 +0100
+++ policy-1.22/file_contexts/program/mta.fc	2005-03-12 01:49:55.546450608 +0100
@@ -5,6 +5,7 @@
 /etc/aliases\.db	--	system_u:object_r:etc_aliases_t
 /var/spool/mail(/.*)?		system_u:object_r:mail_spool_t
 /var/mail(/.*)?			system_u:object_r:mail_spool_t
+/etc/mail(/.*)?			system_u:object_r:etc_mail_t
 ifdef(`postfix.te', `', `
 /usr/sbin/sendmail.postfix --	system_u:object_r:sendmail_exec_t
 /var/spool/postfix(/.*)?		system_u:object_r:mail_spool_t
--- policy-1.18/file_contexts/program/postfix.fc.orig	2004-10-13 22:16:18.000000000 +0200
+++ policy-1.18/file_contexts/program/postfix.fc	2004-11-04 21:55:49.397515200 +0100
@@ -1,10 +1,6 @@
 # postfix
-/etc/postfix(/.*)?		system_u:object_r:postfix_etc_t
-ifdef(`distro_redhat', `
-/etc/postfix/aliases.*		system_u:object_r:etc_aliases_t
-')
-/etc/postfix/postfix-script.* -- system_u:object_r:postfix_exec_t
-/etc/postfix/prng_exch	--	system_u:object_r:postfix_prng_t
+/etc/mail/postfix-script.* -- system_u:object_r:postfix_exec_t
+/etc/mail/prng_exch	--	system_u:object_r:postfix_prng_t
 /usr/lib(exec)?/postfix/.*	--	system_u:object_r:postfix_exec_t
 /usr/lib(exec)?/postfix/cleanup --	system_u:object_r:postfix_cleanup_exec_t
 /usr/lib(exec)?/postfix/local	--	system_u:object_r:postfix_local_exec_t
@@ -25,7 +21,7 @@
 /usr/sbin/postlog	--	system_u:object_r:postfix_master_exec_t
 /usr/sbin/postmap	--	system_u:object_r:postfix_map_exec_t
 /usr/sbin/postqueue	--	system_u:object_r:postfix_postqueue_exec_t
-/usr/sbin/postsuper	--	system_u:object_r:postfix_master_exec_t
+/usr/sbin/postsuper	--	system_u:object_r:postfix_postsuper_exec_t
 /usr/sbin/rmail		--	system_u:object_r:sendmail_exec_t
 /usr/sbin/sendmail.postfix --	system_u:object_r:sendmail_exec_t
 /var/spool/postfix(/[^/]+)?	system_u:object_r:postfix_spool_t
--- policy-1.20/file_contexts/program/sendmail.fc.orig	2004-12-02 14:31:53.000000000 +0100
+++ policy-1.20/file_contexts/program/sendmail.fc	2005-01-08 18:16:21.982327544 +0100
@@ -1,5 +1,4 @@
 # sendmail
-/etc/mail(/.*)?				system_u:object_r:etc_mail_t
 /var/log/sendmail\.st		--	system_u:object_r:sendmail_log_t
 /var/log/mail(/.*)?			system_u:object_r:sendmail_log_t
 /var/run/sendmail\.pid		--	system_u:object_r:sendmail_var_run_t
--- policy-1.22/domains/program/unused/mta.te.orig	2005-02-16 20:40:15.000000000 +0100
+++ policy-1.22/domains/program/unused/mta.te	2005-03-12 01:50:45.860801664 +0100
@@ -83,3 +83,6 @@
 ')
 allow system_mail_t etc_runtime_t:file { getattr read };
 allow system_mail_t urandom_device_t:chr_file read;
+
+# etc_mail_t is the type of /etc/mail.
+type etc_mail_t, file_type, sysadmfile, usercanread;
--- policy-1.10/domains/program/unused/pppd.te.wiget	2004-04-05 19:13:55.000000000 +0200
+++ policy-1.10/domains/program/unused/pppd.te	2004-04-13 22:15:30.436468077 +0200
@@ -42,8 +42,8 @@
 allow pppd_t pppd_secret_t:file r_file_perms;
 
 ifdef(`postfix.te', `
-allow pppd_t postfix_etc_t:dir search;
-allow pppd_t postfix_etc_t:file r_file_perms;
+allow pppd_t etc_mail_t:dir search;
+allow pppd_t etc_mail_t:file r_file_perms;
 allow pppd_t postfix_master_exec_t:file read;
 allow postfix_postqueue_t pppd_t:fd use;
 allow postfix_postqueue_t pppd_t:process sigchld;
--- policy-1.20/domains/program/unused/postfix.te.orig	2005-01-05 17:38:40.000000000 +0100
+++ policy-1.20/domains/program/unused/postfix.te	2005-01-08 18:17:37.279880576 +0100
@@ -8,8 +8,6 @@
 # Type for files created during execution of postfix.
 type postfix_var_run_t, file_type, sysadmfile, pidfile;
 
-type postfix_etc_t, file_type, sysadmfile;
-typealias postfix_etc_t alias etc_postfix_t;
 type postfix_exec_t, file_type, sysadmfile, exec_type;
 type postfix_public_t, file_type, sysadmfile;
 type postfix_private_t, file_type, sysadmfile;
@@ -20,6 +18,7 @@
 
 # postfix needs this for newaliases
 allow { system_mail_t sysadm_mail_t } tmp_t:dir getattr;
+allow { system_mail_t sysadm_mail_t } etc_mail_t:file rw_file_perms;
 
 #################################
 #
@@ -28,13 +27,13 @@
 # postfix_$1_exec_t is the type of the postfix_$1 executables.
 #
 define(`postfix_domain', `
-daemon_core_rules(postfix_$1, `$2')
+daemon_base_domain(postfix_$1, `$2', `nosysadm')
 allow postfix_$1_t self:process setpgid;
 allow postfix_$1_t postfix_master_t:process sigchld;
 allow postfix_master_t postfix_$1_t:process signal;
 
-allow postfix_$1_t { etc_t postfix_etc_t postfix_spool_t }:dir r_dir_perms;
-allow postfix_$1_t postfix_etc_t:file r_file_perms;
+allow postfix_$1_t { etc_t etc_mail_t postfix_spool_t }:dir r_dir_perms;
+allow postfix_$1_t etc_mail_t:file r_file_perms;
 read_locale(postfix_$1_t)
 allow postfix_$1_t etc_t:file { getattr read };
 allow postfix_$1_t self:unix_dgram_socket create_socket_perms;
@@ -96,7 +95,7 @@
 dontaudit postfix_master_t selinux_config_t:dir search;
 can_exec({ sysadm_mail_t system_mail_t }, postfix_master_exec_t)
 ifdef(`distro_redhat', `
-file_type_auto_trans({ sysadm_mail_t system_mail_t postfix_master_t }, postfix_etc_t, etc_aliases_t)
+file_type_auto_trans({ sysadm_mail_t system_mail_t postfix_master_t }, etc_mail_t, etc_aliases_t)
 ', `
 file_type_auto_trans({ sysadm_mail_t system_mail_t }, etc_t, etc_aliases_t)
 ')
@@ -140,7 +139,7 @@
 # for ls to get the current context
 allow postfix_master_t self:file { getattr read };
 ifdef(`direct_sysadm_daemon', `
-allow postfix_master_t postfix_etc_t:file rw_file_perms;
+allow postfix_master_t etc_mail_t:file rw_file_perms;
 allow postfix_master_t devpts_t:dir search;
 ')
 
@@ -226,8 +225,8 @@
 allow postfix_cleanup_t self:process setrlimit;
 
 allow user_mail_domain postfix_spool_t:dir r_dir_perms;
-allow user_mail_domain postfix_etc_t:dir r_dir_perms;
-allow { user_mail_domain initrc_t } postfix_etc_t:file r_file_perms;
+allow user_mail_domain etc_mail_t:dir r_dir_perms;
+allow { user_mail_domain initrc_t } etc_mail_t:file r_file_perms;
 allow user_mail_domain self:capability dac_override;
 
 define(`postfix_user_domain', `
@@ -335,7 +334,7 @@
 domain_auto_trans(postfix_pipe_t, procmail_exec_t, procmail_t)
 ')
 ifdef(`sendmail.te', `
-allow sendmail_t postfix_etc_t:dir search;
+allow sendmail_t etc_mail_t:dir search;
 ')
 
 # Program for creating database files
@@ -343,7 +342,7 @@
 base_file_read_access(postfix_map_t)
 allow postfix_map_t { etc_t etc_runtime_t }:{ file lnk_file } { getattr read };
 tmp_domain(postfix_map)
-create_dir_file(postfix_map_t, postfix_etc_t)
+create_dir_file(postfix_map_t, etc_mail_t)
 allow postfix_map_t self:unix_stream_socket create_stream_socket_perms;
 dontaudit postfix_map_t proc_t:dir { getattr read search };
 dontaudit postfix_map_t local_login_t:fd use;
@@ -355,3 +354,29 @@
 can_network_server(postfix_map_t)
 allow postfix_local_t mail_spool_t:dir { remove_name };
 allow postfix_local_t mail_spool_t:file { unlink };
+
+# postsuper (incl. -d) for sysadm
+postfix_user_domain(postsuper)
+can_exec(postfix_master_t, postfix_postsuper_exec_t)
+domain_auto_trans(sysadm_t, postfix_postsuper_exec_t, postfix_postsuper_t)
+allow postfix_postsuper_t postfix_spool_t:dir rw_dir_perms;
+allow postfix_postsuper_t postfix_spool_t:file { unlink rename getattr };
+allow postfix_postsuper_t postfix_spool_bounce_t:dir rw_dir_perms;
+allow postfix_postsuper_t postfix_spool_bounce_t:file { unlink rename getattr };
+allow postfix_postsuper_t postfix_spool_maildrop_t:dir rw_dir_perms;
+allow postfix_postsuper_t postfix_spool_maildrop_t:file { unlink rename getattr };
+allow postfix_postsuper_t sysadm_devpts_t:chr_file { read write getattr };
+allow postfix_postsuper_t self:capability { setgid setuid };
+
+# because of IPv6 patch using getifaddrs(), which uses netlink
+allow { postfix_bounce_t postfix_cleanup_t postfix_local_t postfix_master_t postfix_pickup_t postfix_postdrop_t postfix_postqueue_t postfix_postsuper_t postfix_qmgr_t postfix_showq_t postfix_smtp_t postfix_smtpd_t } self:netlink_socket { create bind read write };
+allow { postfix_bounce_t postfix_cleanup_t postfix_local_t postfix_master_t postfix_pickup_t postfix_postdrop_t postfix_postqueue_t postfix_postsuper_t postfix_qmgr_t postfix_showq_t postfix_smtp_t postfix_smtpd_t } self:capability { net_admin };
+ 
+allow postfix_master_t sysadm_devpts_t:chr_file { read write getattr };
+
+# allow mailq
+allow { system_mail_t sysadm_mail_t staff_mail_t } self:netlink_socket { create bind read write };
+allow { system_mail_t sysadm_mail_t staff_mail_t } self:capability { net_admin };
+
+# maybe this should be applied for mta_delivery_agent in general?
+rw_dir_create_file(postfix_local_t, mail_spool_t)
--- policy-1.20/domains/program/unused/sendmail.te.orig	2005-01-03 22:31:18.000000000 +0100
+++ policy-1.20/domains/program/unused/sendmail.te	2005-01-08 18:18:10.537824600 +0100
@@ -13,9 +13,6 @@
 # daemon started by the init rc scripts.
 #
 
-# etc_mail_t is the type of /etc/mail.
-type etc_mail_t, file_type, sysadmfile, usercanread;
-
 daemon_domain(sendmail, `, nscd_client_domain, mta_delivery_agent, mail_server_domain, mail_server_sender', nosysadm)
 
 tmp_domain(sendmail)
