--- policy-1.4.2/types/file.te.rhat	2003-12-22 21:46:16.000000000 -0500
+++ policy-1.4.2/types/file.te	2003-12-23 15:40:23.710077390 -0500
@@ -112,10 +112,12 @@
 
 #
 # etc_aliases_t is the type of the aliases database.
-# sendmail needs write access to these files.
 #
 type etc_aliases_t, file_type, sysadmfile;
 
+# bluetooth_conf_t is the type of the /etc/bluetooth dir.
+type bluetooth_conf_t, file_type, sysadmfile;
+
 # resolv_conf_t is the type of the /etc/resolv.conf file.
 # pump needs write access to this file.
 type resolv_conf_t, file_type, sysadmfile;
--- policy-1.4.2/Makefile.rhat	2003-12-06 12:17:04.000000000 -0500
+++ policy-1.4.2/Makefile	2003-12-23 15:39:36.516518780 -0500
@@ -19,7 +19,7 @@
 PREFIX = /usr
 BINDIR = $(PREFIX)/bin
 SBINDIR = $(PREFIX)/sbin
-LOADPOLICY  = /sbin/load_policy
+LOADPOLICY  = $(SBINDIR)/load_policy
 CHECKPOLICY = $(BINDIR)/checkpolicy
 SETFILES = $(SBINDIR)/setfiles
 
@@ -88,7 +88,7 @@
 	mkdir -p tmp
 	cat $^ > $@
 
-install-src: 
+install-src: $(APPFILES)
 	rm -rf $(SRCINSTALLDIR)/policy.old
 	-mv $(SRCINSTALLDIR)/policy $(SRCINSTALLDIR)/policy.old
 	mkdir -p $(SRCINSTALLDIR)/policy
