--- policy-1.22/file_contexts/program/mta.fc.orig	2005-02-16 20:40:16.000000000 +0100
+++ policy-1.22/file_contexts/program/mta.fc	2005-03-12 01:49:55.546450608 +0100
@@ -5,6 +5,7 @@
 /etc/aliases\.db	--	system_u:object_r:etc_aliases_t
 /var/spool/mail(/.*)?		system_u:object_r:mail_spool_t
 /var/mail(/.*)?			system_u:object_r:mail_spool_t
+/etc/mail(/.*)?			system_u:object_r:etc_mail_t
 ifdef(`postfix.te', `', `
 /usr/sbin/sendmail.postfix --	system_u:object_r:sendmail_exec_t
 /var/spool/postfix(/.*)?		system_u:object_r:mail_spool_t
--- policy-1.24/file_contexts/program/postfix.fc.orig	2005-05-23 21:20:04.000000000 +0200
+++ policy-1.24/file_contexts/program/postfix.fc	2005-07-02 00:39:33.974529480 +0200
@@ -1,5 +1,4 @@
 # postfix
-/etc/postfix(/.*)?		system_u:object_r:postfix_etc_t
 ifdef(`distro_redhat', `
 /etc/postfix/aliases.*		system_u:object_r:etc_aliases_t
 /usr/libexec/postfix/.*	--	system_u:object_r:postfix_exec_t
@@ -26,8 +25,8 @@
 /usr/lib/postfix/bounce	--	system_u:object_r:postfix_bounce_exec_t
 /usr/lib/postfix/pipe	--	system_u:object_r:postfix_pipe_exec_t
 ')
-/etc/postfix/postfix-script.* -- system_u:object_r:postfix_exec_t
-/etc/postfix/prng_exch	--	system_u:object_r:postfix_prng_t
+/etc/mail/postfix-script.* -- system_u:object_r:postfix_exec_t
+/etc/mail/prng_exch	--	system_u:object_r:postfix_prng_t
 /usr/sbin/postalias	--	system_u:object_r:postfix_master_exec_t
 /usr/sbin/postcat	--	system_u:object_r:postfix_master_exec_t
 /usr/sbin/postdrop	--	system_u:object_r:postfix_postdrop_exec_t
@@ -37,7 +36,7 @@
 /usr/sbin/postlog	--	system_u:object_r:postfix_master_exec_t
 /usr/sbin/postmap	--	system_u:object_r:postfix_map_exec_t
 /usr/sbin/postqueue	--	system_u:object_r:postfix_postqueue_exec_t
-/usr/sbin/postsuper	--	system_u:object_r:postfix_master_exec_t
+/usr/sbin/postsuper	--	system_u:object_r:postfix_postsuper_exec_t
 /usr/sbin/rmail		--	system_u:object_r:sendmail_exec_t
 /usr/sbin/sendmail.postfix --	system_u:object_r:sendmail_exec_t
 /var/spool/postfix(/.*)?	system_u:object_r:postfix_spool_t
--- policy-1.20/file_contexts/program/sendmail.fc.orig	2004-12-02 14:31:53.000000000 +0100
+++ policy-1.20/file_contexts/program/sendmail.fc	2005-01-08 18:16:21.982327544 +0100
@@ -1,5 +1,4 @@
 # sendmail
-/etc/mail(/.*)?				system_u:object_r:etc_mail_t
 /var/log/sendmail\.st		--	system_u:object_r:sendmail_log_t
 /var/log/mail(/.*)?			system_u:object_r:sendmail_log_t
 /var/run/sendmail\.pid		--	system_u:object_r:sendmail_var_run_t
--- policy-1.24/domains/program/unused/pppd.te.orig	2005-05-23 21:20:02.000000000 +0200
+++ policy-1.24/domains/program/unused/pppd.te	2005-07-02 00:40:23.013074488 +0200
@@ -44,8 +44,8 @@
 allow pppd_t pppd_secret_t:file r_file_perms;
 
 ifdef(`postfix.te', `
-allow pppd_t postfix_etc_t:dir search;
-allow pppd_t postfix_etc_t:file r_file_perms;
+allow pppd_t etc_mail_t:dir search;
+allow pppd_t etc_mail_t:file r_file_perms;
 allow pppd_t postfix_master_exec_t:file { getattr read };
 allow postfix_postqueue_t pppd_t:fd use;
 allow postfix_postqueue_t pppd_t:process sigchld;
--- policy-1.28/domains/program/unused/postfix.te.orig	2005-11-17 08:51:38.000000000 +0100
+++ policy-1.28/domains/program/unused/postfix.te	2005-12-22 21:59:16.635939184 +0100
@@ -8,7 +8,6 @@
 # Type for files created during execution of postfix.
 type postfix_var_run_t, file_type, sysadmfile, pidfile;
 
-type postfix_etc_t, file_type, sysadmfile;
 type postfix_exec_t, file_type, sysadmfile, exec_type;
 type postfix_public_t, file_type, sysadmfile;
 type postfix_private_t, file_type, sysadmfile;
@@ -19,6 +18,7 @@
 
 # postfix needs this for newaliases
 allow { system_mail_t sysadm_mail_t } tmp_t:dir getattr;
+allow { system_mail_t sysadm_mail_t } etc_mail_t:file rw_file_perms;
 
 #################################
 #
@@ -27,13 +27,13 @@
 # postfix_$1_exec_t is the type of the postfix_$1 executables.
 #
 define(`postfix_domain', `
-daemon_core_rules(postfix_$1, `$2')
+daemon_base_domain(postfix_$1, `$2', `nosysadm')
 allow postfix_$1_t self:process setpgid;
 allow postfix_$1_t postfix_master_t:process sigchld;
 allow postfix_master_t postfix_$1_t:process signal;
 
-allow postfix_$1_t { etc_t postfix_etc_t postfix_spool_t }:dir r_dir_perms;
-allow postfix_$1_t postfix_etc_t:file r_file_perms;
+allow postfix_$1_t { etc_t etc_mail_t postfix_spool_t }:dir r_dir_perms;
+allow postfix_$1_t etc_mail_t:file r_file_perms;
 read_locale(postfix_$1_t)
 allow postfix_$1_t etc_t:file { getattr read };
 allow postfix_$1_t self:unix_dgram_socket create_socket_perms;
@@ -88,7 +88,7 @@
 ifdef(`targeted_policy', `', `
 role_transition sysadm_r postfix_master_exec_t system_r;
 ')
-allow postfix_master_t postfix_etc_t:file rw_file_perms;
+allow postfix_master_t etc_mail_t:file rw_file_perms;
 dontaudit postfix_master_t admin_tty_type:chr_file { read write };
 allow postfix_master_t devpts_t:dir search;
 
@@ -113,7 +113,7 @@
 can_exec({ sysadm_mail_t system_mail_t }, postfix_master_exec_t)
 ifdef(`distro_redhat', `
 # compatability for old default main.cf
-file_type_auto_trans({ sysadm_mail_t system_mail_t postfix_master_t }, postfix_etc_t, etc_aliases_t)
+file_type_auto_trans({ sysadm_mail_t system_mail_t postfix_master_t }, etc_mail_t, etc_aliases_t)
 # for newer main.cf that uses /etc/aliases
 file_type_auto_trans(postfix_master_t, etc_t, etc_aliases_t)
 ')
@@ -240,8 +240,8 @@
 allow postfix_cleanup_t self:process setrlimit;
 
 allow user_mail_domain postfix_spool_t:dir r_dir_perms;
-allow user_mail_domain postfix_etc_t:dir r_dir_perms;
-allow { user_mail_domain initrc_t } postfix_etc_t:file r_file_perms;
+allow user_mail_domain etc_mail_t:dir r_dir_perms;
+allow { user_mail_domain initrc_t } etc_mail_t:file r_file_perms;
 allow user_mail_domain self:capability dac_override;
 
 define(`postfix_user_domain', `
@@ -351,7 +351,7 @@
 domain_auto_trans(postfix_pipe_t, procmail_exec_t, procmail_t)
 ')
 ifdef(`sendmail.te', `
-r_dir_file(sendmail_t, postfix_etc_t)
+r_dir_file(sendmail_t, etc_mail_t)
 allow sendmail_t postfix_spool_t:dir search;
 ')
 
@@ -360,7 +360,7 @@
 base_file_read_access(postfix_map_t)
 allow postfix_map_t { etc_t etc_runtime_t }:{ file lnk_file } { getattr read };
 tmp_domain(postfix_map)
-create_dir_file(postfix_map_t, postfix_etc_t)
+create_dir_file(postfix_map_t, etc_mail_t)
 allow postfix_map_t self:unix_stream_socket create_stream_socket_perms;
 dontaudit postfix_map_t proc_t:dir { getattr read search };
 dontaudit postfix_map_t local_login_t:fd use;
@@ -371,3 +371,29 @@
 dontaudit postfix_map_t var_t:dir search;
 can_network_server(postfix_map_t)
 allow postfix_map_t port_type:tcp_socket name_connect;
+
+# postsuper (incl. -d) for sysadm
+postfix_user_domain(postsuper)
+can_exec(postfix_master_t, postfix_postsuper_exec_t)
+domain_auto_trans(sysadm_t, postfix_postsuper_exec_t, postfix_postsuper_t)
+allow postfix_postsuper_t postfix_spool_t:dir rw_dir_perms;
+allow postfix_postsuper_t postfix_spool_t:file { unlink rename getattr };
+allow postfix_postsuper_t postfix_spool_bounce_t:dir rw_dir_perms;
+allow postfix_postsuper_t postfix_spool_bounce_t:file { unlink rename getattr };
+allow postfix_postsuper_t postfix_spool_maildrop_t:dir rw_dir_perms;
+allow postfix_postsuper_t postfix_spool_maildrop_t:file { unlink rename getattr };
+allow postfix_postsuper_t sysadm_devpts_t:chr_file { read write getattr };
+allow postfix_postsuper_t self:capability { setgid setuid };
+
+# because of IPv6 patch using getifaddrs(), which uses netlink
+allow { postfix_bounce_t postfix_cleanup_t postfix_local_t postfix_master_t postfix_pickup_t postfix_postdrop_t postfix_postqueue_t postfix_postsuper_t postfix_qmgr_t postfix_showq_t postfix_smtp_t postfix_smtpd_t } self:netlink_socket { create bind read write };
+allow { postfix_bounce_t postfix_cleanup_t postfix_local_t postfix_master_t postfix_pickup_t postfix_postdrop_t postfix_postqueue_t postfix_postsuper_t postfix_qmgr_t postfix_showq_t postfix_smtp_t postfix_smtpd_t } self:capability { net_admin };
+ 
+allow postfix_master_t sysadm_devpts_t:chr_file { read write getattr };
+
+# allow mailq
+allow { system_mail_t sysadm_mail_t staff_mail_t } self:netlink_socket { create bind read write };
+allow { system_mail_t sysadm_mail_t staff_mail_t } self:capability { net_admin };
+
+# maybe this should be applied for mta_delivery_agent in general?
+rw_dir_create_file(postfix_local_t, mail_spool_t)
