--- policy-1.24/domains/program/getty.te.orig	2005-04-29 21:04:29.000000000 +0200
+++ policy-1.24/domains/program/getty.te	2005-07-02 00:49:17.075884600 +0200
@@ -24,7 +24,7 @@
 allow getty_t self:unix_stream_socket create_socket_perms;
 
 # Use capabilities.
-allow getty_t self:capability { dac_override chown sys_resource sys_tty_config };
+allow getty_t self:capability { dac_override chown sys_resource sys_tty_config fowner fsetid };
 
 read_locale(getty_t)
 
@@ -48,6 +48,7 @@
 allow getty_t fs_t:filesystem getattr;
 
 lock_domain(getty)
+rw_dir_create_file(getty_t, var_log_t)
 r_dir_file(getty_t, sysfs_t)
 # for mgetty
 var_run_domain(getty)
