--- policy-1.4.10/Makefile.rhat	2004-01-26 09:04:54.000000000 -0500
+++ policy-1.4.10/Makefile	2004-01-28 14:06:00.246302943 -0500
@@ -96,7 +96,7 @@
 	m4 $(M4PARAM) -Imacros -s $^ > $@.tmp
 	mv $@.tmp $@
 
-install-src: 
+install-src: $(APPFILES) $(FC)
 	rm -rf $(SRCINSTALLDIR)/policy.old
 	-mv $(SRCINSTALLDIR)/policy $(SRCINSTALLDIR)/policy.old
 	mkdir -p $(SRCINSTALLDIR)/policy
@@ -116,7 +116,7 @@
 
 $(FC): $(FCFILES) domains/program domains/misc file_contexts/program file_contexts/misc users
 	cat $(FCFILES) > $@.tmp
-	@/usr/sbin/selinux-make-homedir $@.tmp  > $@.homedirtmp
+	@/usr/sbin/genhomedircon $@.tmp  > $@.homedirtmp
 	cat $@.homedirtmp >> $@.tmp
 	@rm $@.homedirtmp
 	mv $@.tmp $@
